TemplePlus/action_sequence.cpp

```diff
@@ -4053,6 +4068,45 @@ int ActnSeqReplacements::SeqRenderAttack(D20Actn* d20a, UiIntgameTurnbasedFlags
 	return 0;
 }
 
+ActionErrorCode ActnSeqReplacements::ActionCostReload(D20Actn *d20, TurnBasedStatus *tbStat, ActionCostPacket *acp)
+{
+	// free action by default
+	acp->hourglassCost = 0;
+
+	if (d20->d20Caf & D20CAF_NEED_ANIM_COMPLETED) return AEC_OK;
+	if (!combatSys.isCombatActive()) return AEC_OK;
+
+	auto weapon = inventory.ItemWornAt(d20->d20APerformer, EquipSlot::WeaponPrimary);
+	if (!weapon) return AEC_OK;
+
+	// TODO: rapid reload is actually a per-weapon feat per the SRD
+	bool rapid = feats.HasFeatCountByClass(d20->d20APerformer, FEAT_RAPID_RELOAD);
+
+	// costs for reloading heavy/repeating crossbows based on strict rules status
+	int fastCost = config.stricterRulesEnforcement ? 1 : 0; // move or free
+	int slowCost = config.stricterRulesEnforcement ? 4 : 1; // full round or move
```

Comment:
Costs instead of integers would be more readable as constants...

Reply:
Oh, and I can do this, I think. But `0` is actually called `Null` I think, which is a little weird. I think maybe because `Free` was the ToEE code name for what later came to be known as `Swift` actions (the ones that are quick like a free action, but you're only allowed to do one per round).